app/models/manageiq/providers/openstack/infra_manager/host.rb

```diff
@@ -83,6 +84,178 @@ def authentication_status
+                      },
+                      {
+                        :component  => "password-field",
+                        :id         => "authentications.ssh_keypair.password",
+                        :name       => "authentications.ssh_keypair.password",
+                        :label      => _("Password"),
+                        :type       => "password",
+                        :isRequired => true,
+                        :validate   => [{:type => "required"}],
+                      },
+                    ],
+                  },
+                ],
+              },
+              {
+                :component => 'tab-item',
+                :id        => 'ws-tab',
+                :name      => 'ws-tab',
+                :title     => _('Web Service'),
+                :fields    => [
+                  {
+                    :component  => 'validate-host-credentials',
+                    :id         => 'endpoints.ws.valid',
+                    :name       => 'endpoints.ws.valid',
+                    :skipSubmit => true,
+                    :isRequired => true,
+                    :fields     => [
+                      {
+                        :component  => "text-field",
+                        :id         => "authentications.ws.userid",
+                        :name       => "authentications.ws.userid",
+                        :label      => _("Username"),
+                        :isRequired => true,
+                        :validate   => [{:type => "required"}],
+                      },
+                      {
+                        :component  => "password-field",
+                        :id         => "authentications.ws.password",
+                        :name       => "authentications.ws.password",
+                        :label      => _("Password"),
+                        :type       => "password",
+                        :isRequired => true,
+                        :validate   => [{:type => "required"}],
+                        :helperText => _('Used for access to Web Services.')
+                      },
+                    ],
+                  },
+                ],
+              },
+            ]
+          ]
+        },
+      ]
+    }
+  end
+
+  def verify_credentials(auth_type = nil, options = {})
+    raise MiqException::MiqHostError, "No credentials defined" if missing_credentials?(auth_type)
+    raise MiqException::MiqHostError, "Logon to platform [#{os_image_name}] not supported" if auth_type.to_s != 'ipmi' && os_image_name !~ /linux_*/
+
+    case auth_type.to_s
+    when 'remote', 'default', 'ssh_keypair' then verify_credentials_with_ssh(auth_type, options)
+    when 'ws'                               then verify_credentials_with_ws(auth_type)
+    when 'ipmi'                             then verify_credentials_with_ipmi(auth_type)
+    else
+      verify_credentials_with_ws(auth_type)
+    end
+
+    true
+  end
+
```

Comment:
Believe @kbrock deleted this because it is now a duplicate of the one in the `::Host` base class, https://github.com/ManageIQ/manageiq-providers-openstack/pull/839
```suggestion
```